pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -116,7 +116,7 @@ pub mod pallet {
 		type Message: LPEncoding + Clone + Debug + PartialEq + MaxEncodedLen + TypeInfo + FullCodec;
 
 		/// The message router type that is stored for each domain.
-		type Router: DomainRouter<Sender = Self::AccountId>
+		type Router: DomainRouter<Sender = Self::AccountId, Hash = Self::Hash>
```

Comment:
Are we sure the Hash used by Solidity is the same as the Hash used by `frame_system::Config`?

Reply:
I don't think that these hashes have to match, mainly because these are specific to the LP gateway here.

Reply:
We can then force the type to use the `[u8; 32]` (the `Proof` alias) as the standard dictates so

Reply:
This is not the message proof/hash. This is simply a hash for the router that we use in storage to avoid unnecessary storage of routers when queueing message and so on.

Reply:
Ok, is it like an Id of the router, right?

Reply:
Yes. 

And the idea of having it as a hash that we choose how to calculate allows us to change/update a router without necessarily changing its hash. If we have a router with some invalid gas limit, we can replace it with one with a correct gas limit but with the same hash. Was thinking that this might make our lives easier if we are to retry the execution of a failed message. WDYT?

Reply:
TBH, it all depends on how this will be used. 

However taking into account the last thing Jeroen has said about "votes" and how each time a router is updated we need invalidate all. Maybe just allowing the user to set all routers each time and use the index arrays is enough. But whatever you think it's easier/simpler!

Reply:
Array indexes might be problematic if the new set of routers has a different ordering. I would still prefer to have hashes tbh.

Reply:
Note that because all router information is reset each time you set new routers, there is no problem there. There is nothing wrong with the hashes, but it's true that it requires a bit more code (in the router side) and a bit more storage on-chain because a hash is bigger than a number.

Reply:
But maybe the hashes makes sense because you need them later to know which one to choose in order to send the message 👍🏻